server/src/main/java/org/opensearch/action/support/IndicesOptions.java

```diff
@@ -384,6 +384,11 @@ public static IndicesOptions fromOptions(
         }
         if (expandToClosedIndices) {
             wildcards.add(WildcardStates.CLOSED);
+            if (forbidClosedIndices == true) {
```

Comment:
@srikanthpadakanti looked closely into the issue, I believe this validation logic belongs to `IndexNameExpressionResolver::concreteIndices` where the wildcards (index options) are resolved into concrete indices.

Reply:
@reta Not just IndicesOptionsTests.java but there are bunch of there tests are failing with the same exception. 
https://build.ci.opensearch.org/job/gradle-check/36530/console